webcomponents-loader.js

```diff
@@ -27,48 +85,41 @@
     polyfills.push('ce');
   }
   // NOTE: any browser that does not have template or ES6 features
-  // must load the full suite (called `lite` for legacy reasons) of polyfills.
+  // must load the full suite of polyfills.
   if (!('content' in document.createElement('template')) || !window.Promise || !Array.from ||
     // Edge has broken fragment cloning which means you cannot clone template.content
     !(document.createDocumentFragment().cloneNode() instanceof DocumentFragment)) {
-    polyfills = ['lite'];
+    polyfills = ['sd-ce-pf'];
   }
 
   if (polyfills.length) {
     var script = document.querySelector('script[src*="' + name +'"]');
     var newScript = document.createElement('script');
     // Load it from the right place.
-    var replacement = 'webcomponents-' + polyfills.join('-') + '.js';
+    var replacement = 'bundles/webcomponents-' + polyfills.join('-') + '.js';
     var url = script.src.replace(name, replacement);
     newScript.src = url;
-    // NOTE: this is required to ensure the polyfills are loaded before
-    // *native* html imports load on older Chrome versions. This *is* CSP
-    // compliant since CSP rules must have allowed this script to run.
-    // In all other cases, this can be async.
-    if (document.readyState === 'loading' && ('import' in document.createElement('link'))) {
+    // if readyState is 'loading', this script is synchronous
+    if (document.readyState === 'loading') {
+      // make sure custom elements are batched whenever parser gets to the injected script
+      newScript.setAttribute('onload', 'window.WebComponents._batchCustomElements()');
       document.write(newScript.outerHTML);
+      document.addEventListener('DOMContentLoaded', ready);
     } else {
+      newScript.addEventListener('load', function () {
+        asyncReady();
+      });
+      newScript.addEventListener('error', function () {
+        throw new Error('Could not load polyfill bundle' + url);
+      });
       document.head.appendChild(newScript);
     }
   } else {
-    // Ensure `WebComponentsReady` is fired also when there are no polyfills loaded.
-    // however, we have to wait for the document to be in 'interactive' state,
-    // otherwise a rAF may fire before scripts in <body>
-
-    var fire = function() {
-      requestAnimationFrame(function() {
-        window.WebComponents.ready = true;
-        document.dispatchEvent(new CustomEvent('WebComponentsReady', {bubbles: true}));
-      });
-    };
-
-    if (document.readyState !== 'loading') {
-      fire();
+    polyfillsLoaded = true;
+    if (document.readyState === 'complete') {
+      ready()
```

Comment:
We can change `ready` to `fireEvent` here since it's not possible to call `waitFor` before this.

Reply:
Done